src/Datasource/EntityTrait.php

```diff
@@ -283,6 +320,23 @@ public function &get($property)
             throw new InvalidArgumentException('Cannot get an empty property');
         }
 
+        $nested = false;
+        if (strpos($property, '.') > 0 && !array_key_exists($property, $this->_properties)) {
+            list($property, $nested) = explode('.', $property, 2);
+        }
+
+        if ($nested) {
+            if ($this->_properties[$property] instanceof EntityInterface) {
+                $result = $this->_properties[$property]->get($nested);
+                return $result;
+            } elseif (is_array($this->_properties[$property])
+                || ($this->_properties[$property] instanceof \ArrayAccess)
+            ) {
+                $value = Hash::get($this->_properties[$property], $nested);
+                return $value; //@TODO: Or remove this line and let getters engage below?
```

Comment:
Engaging getters/setters seems like it would be least surprising. It also has the nice effect of making

```php
$article->get('author.group.name');
```

into 'syntactic sugar' for

```php
$article->author->group->name;
```